ksqldb-serde/src/test/java/io/confluent/ksql/serde/json/KsqlJsonDeserializerTest.java

```diff
@@ -538,6 +538,40 @@ public void shouldDeserializeDecimalsWithoutStrippingTrailingZeros() {
     assertThat(result, is(new BigDecimal("10.0")));
   }
 
+  @Test
+  public void shouldFixScaleWhenDeserializingDecimalsWithTooSmallAScale() {
+    // Given:
+    final KsqlJsonDeserializer<BigDecimal> deserializer =
+        givenDeserializerForSchema(DecimalUtil.builder(4, 3).build(), BigDecimal.class);
+
+    final byte[] bytes = addMagic("1.1".getBytes(UTF_8));
+
+    // When:
+    final Object result = deserializer.deserialize(SOME_TOPIC, bytes);
+
+    // Then:
+    assertThat(result, is(new BigDecimal("1.100")));
+  }
+
+  @Test
+  public void shouldThrowIfDecimalHasLargerScale() {
```

Comment:
Does ksqlDB always throw when deserializing decimals that don't fit into the specified precision/scale? What about when casting from other numeric types to decimals? There's no way for the user to specify rounding behavior? (Asking in case you know off the top of your head. If not I can poke around the code, not seeing anything in the docs one way or the other.)

Reply:
@agavra can probably answer more definitely than me, as he did the initial work. However, I believe a deserializer _should_ throw if the deserialized value doesn't fit in the ksql decimal scale and precision.  (See `DecimalUtil.ensureFit`). This isn't necessary for some formats, e.g. `AVRO`, as the Avro schema dictates scale and precision, and the ksql schema is driven from the Avro one. 

Casting is supported, and I'm assuming `ensureFit` is being called appropriately.

I don't think there's any way for a user to specify rounding behaviour, though I'm fairly certain it was discussed during the design, but punted to a later feature.

Reply:
it's been a while since i've looked at the code but I agree with what Andy is saying. As far as casting is concerned, we hardcode the scale/precision for casts:
```java
  /**
   * Schema of an INT up-cast to a DECIMAL
   */
  public static final SqlDecimal INT_UPCAST_TO_DECIMAL = SqlDecimal.of(10, 0);

  /**
   * Schema of an BIGINT up-cast to a DECIMAL
   */
  public static final SqlDecimal BIGINT_UPCAST_TO_DECIMAL = SqlDecimal.of(19, 0);
```
And then for `double` we do some other magic